File: android/app/src/main/kotlin/com/renovex/capture/network/AuthApi.kt
package com.renovex.capture.network

import kotlinx.serialization.Serializable
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

/**
 * Mirrors backend/internal/identity/handler.go's JSON contract exactly
 * (Task 1 inspection). The refresh_token cookie itself is never modeled
 * here — OkHttp's CookieJar (see RenovexCookieJar) handles it transparently,
 * matching the same cookie-based flow apps/web/src/lib/api/client.ts uses.
 */
interface AuthApi {
    @POST("/auth/login")
    suspend fun login(@Body body: LoginRequest): Response<AuthResponse>

    @POST("/auth/register")
    suspend fun register(@Body body: RegisterRequest): Response<AuthResponse>

    @POST("/auth/refresh")
    suspend fun refresh(): Response<AuthResponse>

    @POST("/auth/logout")
    suspend fun logout(): Response<Unit>
}

@Serializable
data class LoginRequest(val email: String, val password: String)

@Serializable
data class RegisterRequest(val email: String, val password: String, val companyName: String)

@Serializable
data class AuthResponse(val accessToken: String, val mustChangePassword: Boolean)

File: ios/RenovexCapture/Sources/RenovexCaptureCore/Networking/AuthAPI.swift
import Foundation

/// Mirrors `backend/internal/identity/handler.go`'s JSON contract exactly
/// (Task 1 inspection, reused from the frozen Android provider's
/// `network/AuthApi.kt`). The `refresh_token` cookie itself is never
/// modeled here — `URLSession`'s shared `HTTPCookieStorage` handles it
/// transparently, matching the same cookie-based flow
/// `apps/web/src/lib/api/client.ts` and the Android `RenovexCookieJar` use.
public struct LoginRequestBody: Encodable, Sendable {
    public let email: String
    public let password: String

    public init(email: String, password: String) {
        self.email = email
        self.password = password
    }
}

public struct AuthResponseBody: Decodable, Sendable {
    public let accessToken: String
    public let mustChangePassword: Bool
}
